slang.h

```diff
@@ -2098,6 +2098,21 @@ extern "C"
         SlangReflectionType* const* specializationArgs,
         ISlangBlob**                outDiagnostics);
 
+        /// Get the number of hashed strings
+    SLANG_API SlangUInt spReflection_getHashedStringCount(
```

Comment:
In terms of naming of the API, should this be something like `spReflection_getStaticStringCount` or `_getLiteralStringCount` or something like that? It doesn't seem like this should be restricted to those strings that were explicitly passed to `getStringHash(...)`, but should rather include any/all string literals mentioned in the source.

I'm guessing that right now the name is accurate, because we are only reflecting those strings that have been hashed as part of compilation?

Reply:
Yes - I purposefully only track strings literals passed to getStringHash. I think that is the desirable behavior.

Reply:
That probably depends on the direction we want to take for the bigger-picture feature. I can see two different approaches:

1. Allow string literals to be used and passed around as `String`, and decide that is okay so long as somebody along the call chain eventually uses `getStringHash()`. Implement an IR pass that transforms `String`-type function parameters into `int`-type parameters and then passes around the integers instead of the originals by moving the `getStringHash()` calls to the places where the literals originate. This lets people write `String`-based code and only have to think about hashes at the lowest levels of a `print()` implementation.

2. Expect user code to traffic in a `StringHash` type, and add an implicit conversion from `String` to `StringHash` that uses `getStringHash` as its implementation. User subroutines, etc. are expected to be written in terms of `StringHash` instead of `String` which surfaces the details more, but at least clients of `print()` don't have to think about it because of the implicit conversion.

The choices in this PR are consistent with (2), but not with (1). That's probably fine, but it is the kind of thing we should make sure we document as the POR.

Reply:
Of the two, 2 seems preferable from the point of view of clarity and simplicity of implementation - especially in light of the string literal requirement.

Reply:
I think I agree, but I want to make sure design decisions like this are explicit/apparent for the benefit of anybody coming to the project later.

There’s a mild argument to be made that one could write more reusable code by allowing subroutines to take `String` even if it is limited to literals in practice for some targets (while other targets wouldn’t have that restriction).